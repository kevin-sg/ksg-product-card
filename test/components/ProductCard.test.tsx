import React from 'react';
import '../setupTest';
import { mount } from 'enzyme';

import ProductCard from '../../src/components';
import { product1 } from '../data/products';
import { act } from 'react-dom/test-utils';

describe('ProductCard', () => {
  test('Debe mostrar componente con la imagen del producto', () => {
    const wrapper = mount(
      <ProductCard product={product1}>{() => <h1>Product Card</h1>}</ProductCard>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe incrementar el contador', () => {
    const wrapper = mount(
      <ProductCard product={product1}>
        {({ count, increaseBy }) => (
          <>
            <h1>Product Card</h1>
            <span>{count}</span>
            <button onClick={() => increaseBy(1)}>Increment</button>
          </>
        )}
      </ProductCard>
    );
    expect(wrapper).toMatchSnapshot();

    act(() => {
      (wrapper as any).find('button').prop('onClick')();
    });

    expect(wrapper.find('span').text()).toBe('1');
  });
});
