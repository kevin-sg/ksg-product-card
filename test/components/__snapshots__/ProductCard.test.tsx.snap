// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard Debe incrementar el contador 1`] = `
<div
  className="productCard "
>
  <h1>
    Product Card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;

exports[`ProductCard Debe mostrar componente con la imagen del producto 1`] = `
<div
  className="productCard "
>
  <h1>
    Product Card
  </h1>
</div>
`;
